addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/handler/AVMFritzBaseBridgeHandler.java

```diff
@@ -285,7 +286,28 @@ protected void updateThingFromDevice(Thing thing, AVMFritzBaseModel device) {
                 updateThingChannelState(thing, CHANNEL_BATTERY_LOW, UnDefType.UNDEF);
             } else {
                 updateThingChannelState(thing, CHANNEL_BATTERY_LOW,
-                        HeatingModel.BATTERY_ON.equals(device.getHkr().getBatterylow()) ? OnOffType.ON : OnOffType.OFF);
+                        BATTERY_ON.equals(device.getHkr().getBatterylow()) ? OnOffType.ON : OnOffType.OFF);
+            }
+        }
+        if (device instanceof DeviceModel && device.isAlarmSensor() && ((DeviceModel) device).getAlert() != null) {
+            updateThingChannelState(thing, CHANNEL_CONTACT_STATE,
+                    AlertModel.ON.equals(((DeviceModel) device).getAlert().getState()) ? OpenClosedType.OPEN
+                            : OpenClosedType.CLOSED);
+        }
+        if (device instanceof DeviceModel && device.isButton() && ((DeviceModel) device).getButton() != null) {
+            if (((DeviceModel) device).getButton().getLastpressedtimestamp() == 0) {
+                updateThingChannelState(thing, CHANNEL_LAST_CHANGE, UnDefType.UNDEF);
+            } else {
+                ZoneId zoneId = ZoneId.systemDefault();
+                ZonedDateTime timestamp = ZonedDateTime.ofInstant(
+                        Instant.ofEpochSecond(((DeviceModel) device).getButton().getLastpressedtimestamp()), zoneId);
```

Comment:
`getLastpressedtimestamp` proper naming of the words? Or an api defect?

Reply:
Yes, changing this will lead to an API defect. The JAXB Parser starts complaining if I change it.